beacon-chain/blockchain/service_test.go

```diff
@@ -198,6 +198,69 @@ func TestFaultyStop(t *testing.T) {
 }
 
 func TestProcessingBadBlock(t *testing.T) {
+	ctx := context.Background()
+	config := &database.DBConfig{DataDir: "", Name: "", InMemory: true}
+	db, err := database.NewDB(config)
+	if err != nil {
+		t.Fatalf("could not setup beaconDB: %v", err)
+
+	}
+	endpoint := "ws://127.0.0.1"
+	client := &mockClient{}
+	web3Service, err := powchain.NewWeb3Service(ctx, &powchain.Web3ServiceConfig{Endpoint: endpoint, Pubkey: "", VrcAddr: common.Address{}}, client, client, client)
+	if err != nil {
+		t.Fatalf("unable to set up web3 service: %v", err)
+	}
+	beaconChain, err := NewBeaconChain(db.DB())
+	if err != nil {
+		t.Fatalf("could not register blockchain service: %v", err)
+	}
+	cfg := &Config{
+		BeaconBlockBuf:   0,
+		IncomingBlockBuf: 0,
+		BeaconDB:         db.DB(),
+		Chain:            beaconChain,
+		Web3Service:      web3Service,
+	}
+	chainService, _ := NewChainService(ctx, cfg)
+
+	active := types.NewActiveState(&pb.ActiveState{RecentBlockHashes: [][]byte{{'A'}}}, make(map[*common.Hash]*types.VoteCache))
+	activeStateHash, err := active.Hash()
+	if err != nil {
+		t.Fatalf("Cannot hash active state: %v", err)
+	}
+	chainService.chain.SetActiveState(active)
+
+	crystallized := types.NewCrystallizedState(&pb.CrystallizedState{LastStateRecalc: 10000})
+	crystallizedStateHash, err := crystallized.Hash()
+	if err != nil {
+		t.Fatalf("Cannot hash crystallized state: %v", err)
+	}
+	chainService.chain.SetCrystallizedState(crystallized)
+
+	parentBlock := NewBlock(t, nil)
+	parentHash, _ := parentBlock.Hash()
+
+	block := NewBlock(t, &pb.BeaconBlock{
+		SlotNumber:            2,
+		ActiveStateHash:       activeStateHash[:],
+		CrystallizedStateHash: crystallizedStateHash[:],
+		ParentHash:            parentHash[:],
+		PowChainRef:           []byte("a"),
+	})
+
+	exitRoutine := make(chan bool)
+	go func() {
+		chainService.blockProcessing(chainService.ctx.Done())
+		<-exitRoutine
+	}()
+
+	chainService.incomingBlockChan <- block
```

Comment:
where do you test processing a bad block ? Its basically initializing all the services and then feeding the block into the channel , no testing for block validity,etc

Reply:
Looks like this is a typo from way back. The test should be `TestProcessingBlock`